### PR TITLE
Pr 629

### PR DESCRIPTION
**Pull request by grutt**
# Description

Closes #629 

## Type of change

- [x] Documentation change (pure documentation change)

### CODE REVIEW

**Comment by grutt** [comment]
@akshat2602 vercel build state got botched on your PR for some reason, will be completing the merge here.

File: frontend/docs/pages/home/features/concurrency/cancel-in-progress.mdx [hunk 1]
```diff
@@ -47,6 +71,45 @@ export const myWorkflow: Workflow = {
...
+
+```go
+import "github.com/hatchet-dev/hatchet/pkg/client/types"
+
+type MyUser struct {
+    UserId string `json:"user_id"`
+}
+
+func getConcurrencyKey(ctx worker.HatchetContext) (string, error) {
+  event := &MyEvent{}
```
Comment (abelanger5) [inline frontend/docs/pages/home/features/concurrency/cancel-in-progress.mdx]:
```suggestion
  event := &MyUser{}
```

File: frontend/docs/pages/home/features/concurrency/cancel-in-progress.mdx [hunk 2]
```diff
@@ -47,6 +71,45 @@ export const myWorkflow: Workflow = {
...
+
+}
+
+err = w.RegisterWorkflow(
+    &worker.WorkflowJob{
+        Name: "concurrency-limit-per-user",
+        On: worker.Events("concurrency-test-event"),
+        Description: "This limits concurrency to 1 run at a time per user.",
+        Concurrency: worker.Concurrency(getConcurrencyKey).MaxRuns(10).LimitStrategy(types.CancelInProgress),
+        Steps: []\*worker.WorkflowStep{
```
Comment (abelanger5) [inline frontend/docs/pages/home/features/concurrency/cancel-in-progress.mdx]:
```suggestion
        Steps: []*worker.WorkflowStep{
```

File: frontend/docs/pages/home/features/concurrency/cancel-in-progress.mdx [hunk 3]
```diff
@@ -47,6 +71,45 @@ export const myWorkflow: Workflow = {
 };
 ```
 
+  </Tabs.Tab>
+  <Tabs.Tab>
+
+```go
+import "github.com/hatchet-dev/hatchet/pkg/client/types"
```
Comment (abelanger5) [inline frontend/docs/pages/home/features/concurrency/cancel-in-progress.mdx]: `worker` needs to be imported

File: frontend/docs/pages/home/features/concurrency/round-robin.mdx [hunk 4]
```diff
@@ -47,6 +70,44 @@ export const myWorkflow: Workflow = {
...
+
+```go
+import "github.com/hatchet-dev/hatchet/pkg/client/types"
+
+type MyUser struct {
+  UserId string `json:"user_id"`
+}
+
+func getConcurrencyKey(ctx worker.HatchetContext) (string, error) {
+    event := &MyEvent{}
```
Comment (abelanger5) [inline frontend/docs/pages/home/features/concurrency/round-robin.mdx]: Same comments as cancel in progress

File: frontend/docs/pages/home/features/errors-and-logging.mdx [hunk 5]
```diff
@@ -45,17 +70,36 @@ const myStep: Step<any, any> = async (context: Context<any>) => {
...
 ```
 
+  </Tabs.Tab>
+  <Tabs.Tab>
+
+```go
+err = w.RegisterWorkflow(
+	&worker.WorkflowJob{
+		Name:        "log-demo-workflow",
+    On: worker.Events("user:create"),
```
Comment (abelanger5) [inline frontend/docs/pages/home/features/errors-and-logging.mdx:86]: Formatting needed

File: frontend/docs/pages/home/features/errors-and-logging.mdx [hunk 6]
```diff
@@ -45,17 +70,36 @@ const myStep: Step<any, any> = async (context: Context<any>) => {
...
+		Description: "This is an example workflow with logging.",
+		Steps: []*worker.WorkflowStep{
+			worker.Fn(func(ctx worker.HatchetContext) (result *stepOneOutput, err error) {
+				ctx.log("Starting step execution")
+        result := someOpeartion()
```
Comment (abelanger5) [inline frontend/docs/pages/home/features/errors-and-logging.mdx]:
```suggestion
        result := someOperation()
```

File: frontend/docs/pages/home/features/retries/simple.mdx [hunk 7]
```diff
@@ -19,15 +19,37 @@ This simple retry mechanism can help to mitigate transient failures, such as net
...
+````
+  </Tabs.Tab>
+  <Tabs.Tab>
+```go
+Steps: []*worker.WorkflowStep{
+  worker.Fn(func(ctx worker.HatchetContext) (result *stepOneOutput, err error) {
+    continue
+  }).SetName("step1").SetRetries(3).SetTimeout("30s"),
+},
+````
```
Comment (abelanger5) [inline frontend/docs/pages/home/features/retries/simple.mdx]: Pretty sure the four backticks are unnecessary if you leave a newline in between the `<Tabs.Tab>`

File: frontend/docs/pages/home/features/timeouts.mdx [hunk 8]
```diff
@@ -140,6 +154,26 @@ const myStep: CreateStep<any, any> = {
   },
 };
 ```
+  </Tabs.Tab>
+  <Tabs.Tab>
+```go
+worker.WorkflowJob{
```
Comment (abelanger5) [inline frontend/docs/pages/home/features/timeouts.mdx:160]: Formatting

**Review by abelanger5 — APPROVED** [review]

File: frontend/docs/pages/home/features/retries/simple.mdx (hunk 7 above)
Comment (akshat2602) [reply frontend/docs/pages/home/features/retries/simple.mdx]: Some kind of linter issue, fixing in the next commit

**Comment by akshat2602** [comment]
I've made changes for this but then how do you want to go about merging them? Should I open a PR to the branch PR 629 and then we merge that and then merge this? @grutt

**Comment by grutt** [comment]
Works

**Pull request merged**